src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/dropdowncomponentobject/DropDownComponentObject.java

```diff
@@ -54,25 +55,14 @@ public Boolean apply(WebDriver webDriver) {
   }
 
   public DropDownComponentObject openDropDownWithEntryPoint(final WebElement entryPoint) {
-    driver.manage().timeouts().implicitlyWait(500, TimeUnit.MILLISECONDS);
     try {
-      new WebDriverWait(driver, 20, 2000).until(new ExpectedCondition<Boolean>() {
-        @Override
-        public Boolean apply(WebDriver webDriver) {
-          if (!entryPoint.getAttribute("class").contains("active")) {
-            ((JavascriptExecutor) driver)
-                    .executeScript("$(arguments[0]).trigger('click')", entryPoint);
-            return false;
-          }
-          return true;
-        }
-      });
-    } finally {
-      restoreDeaultImplicitWait();
+      wait.forElementClickable(entryPoint);
+      PageObjectLogging.log("DropdownClickable", "Dropdown is clickable", true, driver);
+      entryPoint.click();
+    } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)